flink-libraries/flink-table/src/main/scala/org/apache/flink/table/runtime/aggregate/AggregateUtil.scala

```diff
@@ -89,6 +95,18 @@ object AggregateUtil {
         aggregationStateType)
     }
   }
+ 
+  def getLowerBoundary(
+      constants: ImmutableList[RexLiteral],
+      lowerBound: RexWindowBound,
+      input: RelNode): Int = {
+    val ref: RexInputRef = lowerBound.getOffset.asInstanceOf[RexInputRef]
+    val index:Int = ref.getIndex
+    val count: Int = input.getRowType.getFieldCount
```

Comment:
The offset must be used to get the right position constants array. If there are more constants in the query, I need to consider the overall field count and subtract the offset to get the right position in the constants array. To be honest, this part is quite obscure, and also on Calcite mailing list the reply I got was quite confusing and wrong (in practice).